lucene/core/src/test/org/apache/lucene/index/TestIndexWriterMergePolicy.java

```diff
@@ -388,6 +388,69 @@ public void testMergeOnCommit() throws IOException {
     dir.close();
   }
 
+  private class TesIndexWriterEventListener implements IndexWriterEventListener {
+    private boolean beginMergeCalled = false;
+    private boolean endMergeCalled = false;
+
+    @Override
+    public void beginMergeOnFullFlush(MergePolicy.OneMerge merge) {
+      beginMergeCalled = true;
+    }
+
+    @Override
+    public void endMergeOnFullFlush(MergePolicy.OneMerge merge) {
+      endMergeCalled = true;
+    }
+
+    public boolean isEventsRecorded() {
+      return beginMergeCalled && endMergeCalled;
+    }
+  }
+
+  // Test basic semantics of merge on commit and events recording invocation
+  public void testMergeOnCommitWithEventListener() throws IOException {
```

Comment:
You might also edit `LuceneTestCase.newIndexWriterConfig` to randomly swap in a `MockIndexWriterEventListener` just to exercise this listener in any tests using that API, which is quite a few.  It could uncover times when we accidentally break something when this listener is invoked ...

Reply:
Makes sense. Added.